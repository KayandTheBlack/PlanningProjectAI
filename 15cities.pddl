(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI cityJ cityK cityL cityM cityN cityO - city
    hotelA_A hotelA_B hotelB_A hotelB_B hotelC_A hotelC_B hotelD_A hotelD_B hotelE_A hotelE_B hotelF_A hotelF_B hotelG_A hotelG_B hotelH_A hotelH_B hotelI_A hotelI_B hotelJ_A hotelJ_B hotelK_A hotelK_B hotelL_A hotelL_B hotelM_A hotelM_B hotelN_A hotelN_B hotelO_A hotelO_B - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelJ_A cityJ)
    (hotelAt hotelJ_B cityJ)
    (hotelAt hotelK_A cityK)
    (hotelAt hotelK_B cityK)
    (hotelAt hotelL_A cityL)
    (hotelAt hotelL_B cityL)
    (hotelAt hotelM_A cityM)
    (hotelAt hotelM_B cityM)
    (hotelAt hotelN_A cityN)
    (hotelAt hotelN_B cityN)
    (hotelAt hotelO_A cityO)
    (hotelAt hotelO_B cityO)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight Origin cityJ)
    (flight Origin cityK)
    (flight Origin cityL)
    (flight Origin cityM)
    (flight Origin cityN)
    (flight Origin cityO)
    (flight cityA cityB)
    (flight cityA cityO)
    (flight cityB cityD)
    (flight cityB cityH)
    (flight cityC cityI)
    (flight cityC cityO)
    (flight cityD cityK)
    (flight cityD cityM)
    (flight cityE cityH)
    (flight cityE cityM)
    (flight cityF cityG)
    (flight cityF cityH)
    (flight cityG cityC)
    (flight cityG cityD)
    (flight cityH cityB)
    (flight cityH cityE)
    (flight cityI cityG)
    (flight cityI cityK)
    (flight cityJ cityC)
    (flight cityJ cityF)
    (flight cityK cityE)
    (flight cityK cityL)
    (flight cityL cityA)
    (flight cityL cityJ)
    (flight cityM cityG)
    (flight cityM cityN)
    (flight cityN cityB)
    (flight cityN cityM)
    (flight cityO cityG)
    (flight cityO cityN)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 2)
    (= (maxDaysInCity) 5)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (stayDuration cityI) 0)
    (= (stayDuration cityJ) 0)
    (= (stayDuration cityK) 0)
    (= (stayDuration cityL) 0)
    (= (stayDuration cityM) 0)
    (= (stayDuration cityN) 0)
    (= (stayDuration cityO) 0)
    (= (totalcost) 0)
    (= (hotelcost hotelA_A) 155)
    (= (hotelcost hotelA_B) 212)
    (= (hotelcost hotelB_A) 161)
    (= (hotelcost hotelB_B) 95)
    (= (hotelcost hotelC_A) 181)
    (= (hotelcost hotelC_B) 232)
    (= (hotelcost hotelD_A) 160)
    (= (hotelcost hotelD_B) 109)
    (= (hotelcost hotelE_A) 65)
    (= (hotelcost hotelE_B) 142)
    (= (hotelcost hotelF_A) 158)
    (= (hotelcost hotelF_B) 87)
    (= (hotelcost hotelG_A) 81)
    (= (hotelcost hotelG_B) 221)
    (= (hotelcost hotelH_A) 150)
    (= (hotelcost hotelH_B) 140)
    (= (hotelcost hotelI_A) 184)
    (= (hotelcost hotelI_B) 217)
    (= (hotelcost hotelJ_A) 121)
    (= (hotelcost hotelJ_B) 237)
    (= (hotelcost hotelK_A) 247)
    (= (hotelcost hotelK_B) 57)
    (= (hotelcost hotelL_A) 115)
    (= (hotelcost hotelL_B) 160)
    (= (hotelcost hotelM_A) 179)
    (= (hotelcost hotelM_B) 53)
    (= (hotelcost hotelN_A) 91)
    (= (hotelcost hotelN_B) 247)
    (= (hotelcost hotelO_A) 67)
    (= (hotelcost hotelO_B) 119)

    (= (flightcost Origin cityA) 0)
    (= (flightcost Origin cityB) 0)
    (= (flightcost Origin cityC) 0)
    (= (flightcost Origin cityD) 0)
    (= (flightcost Origin cityE) 0)
    (= (flightcost Origin cityF) 0)
    (= (flightcost Origin cityG) 0)
    (= (flightcost Origin cityH) 0)
    (= (flightcost Origin cityI) 0)
    (= (flightcost Origin cityJ) 0)
    (= (flightcost Origin cityK) 0)
    (= (flightcost Origin cityL) 0)
    (= (flightcost Origin cityM) 0)
    (= (flightcost Origin cityN) 0)
    (= (flightcost Origin cityO) 0)
    (= (flightcost cityA cityB) 145)
    (= (flightcost cityA cityO) 122)
    (= (flightcost cityB cityD) 131)
    (= (flightcost cityB cityH) 108)
    (= (flightcost cityC cityI) 119)
    (= (flightcost cityC cityO) 115)
    (= (flightcost cityD cityK) 143)
    (= (flightcost cityD cityM) 130)
    (= (flightcost cityE cityH) 174)
    (= (flightcost cityE cityM) 158)
    (= (flightcost cityF cityG) 174)
    (= (flightcost cityF cityH) 134)
    (= (flightcost cityG cityC) 147)
    (= (flightcost cityG cityD) 105)
    (= (flightcost cityH cityB) 105)
    (= (flightcost cityH cityE) 199)
    (= (flightcost cityI cityG) 195)
    (= (flightcost cityI cityK) 192)
    (= (flightcost cityJ cityC) 167)
    (= (flightcost cityJ cityF) 118)
    (= (flightcost cityK cityE) 179)
    (= (flightcost cityK cityL) 116)
    (= (flightcost cityL cityA) 178)
    (= (flightcost cityL cityJ) 144)
    (= (flightcost cityM cityG) 178)
    (= (flightcost cityM cityN) 107)
    (= (flightcost cityN cityB) 147)
    (= (flightcost cityN cityM) 171)
    (= (flightcost cityO cityG) 104)
    (= (flightcost cityO cityN) 116)

  )

  (:goal
    (and
      (>= (citiesVisited) 14)
      (>= (totalDays) 40)
      (>= (totalcost) 100)
      (<= (totalcost) 1000000)
    )
  )

  (:metric 
    minimize (totalcost)
  )

)
