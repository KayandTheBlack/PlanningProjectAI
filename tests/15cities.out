
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


metric established (normalized to minimize): ((1.00*[RF0](TOTALCOST)) - () + 0.00)

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is ((1.00*[RF0](TOTALCOST)) - () + 0.00)

advancing to distance:   20
                         19
                         18
                         17
                         16
                         15
                         14
                         12
                         11
                         10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYA HOTELA_A
        1: FLYANDSTAY CITYA CITYB HOTELB_B
        2: FLYANDSTAY CITYB CITYH HOTELH_B
        3: FLYANDSTAY CITYH CITYE HOTELE_A
        4: INCREASESTAY CITYE HOTELE_A
        5: INCREASESTAY CITYE HOTELE_A
        6: INCREASESTAY CITYB HOTELB_B
        7: INCREASESTAY CITYB HOTELB_B
        8: INCREASESTAY CITYH HOTELH_B
        9: INCREASESTAY CITYB HOTELB_B
       10: INCREASESTAY CITYH HOTELH_B
       11: INCREASESTAY CITYH HOTELH_B
       12: INCREASESTAY CITYA HOTELA_A
       13: FLYANDSTAY CITYE CITYM HOTELM_B
       14: FLYANDSTAY CITYM CITYG HOTELG_A
       15: FLYANDSTAY CITYG CITYD HOTELD_B
       16: FLYANDSTAY CITYD CITYK HOTELK_B
       17: INCREASESTAY CITYE HOTELE_A
       18: INCREASESTAY CITYM HOTELM_B
       19: INCREASESTAY CITYA HOTELA_A
       20: INCREASESTAY CITYA HOTELA_A
       21: INCREASESTAY CITYM HOTELM_B
       22: INCREASESTAY CITYM HOTELM_B
       23: INCREASESTAY CITYK HOTELK_B
       24: INCREASESTAY CITYK HOTELK_B
       25: INCREASESTAY CITYK HOTELK_B
       26: FLYANDSTAY CITYK CITYL HOTELL_A
       27: FLYANDSTAY CITYL CITYJ HOTELJ_A
       28: FLYANDSTAY CITYJ CITYC HOTELC_A
       29: FLYANDSTAY CITYC CITYO HOTELO_A
       30: FLYANDSTAY CITYO CITYN HOTELN_A
     

time spent:    0.00 seconds instantiating 120 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 77 facts and 120 actions
               0.00 seconds creating final representation with 76 relevant facts, 35 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.00 seconds searching, evaluating 715 states, to a max depth of 0
               0.00 seconds total time

